src/pyhf/tensor/pytorch_backend.py

```diff
@@ -245,6 +245,45 @@ def log(self, tensor_in):
     def exp(self, tensor_in):
         return torch.exp(tensor_in)
 
+    def percentile(self, tensor_in, q, axis=None, interpolation="linear"):
+        r"""
+        Compute the :math:`q`-th percentile of the tensor along the specified axis.
+
+        Example:
+
+            >>> import pyhf
+            >>> pyhf.set_backend("pytorch")
+            >>> a = pyhf.tensorlib.astensor([[10, 7, 4], [3, 2, 1]])
+            >>> pyhf.tensorlib.percentile(a, 50)
+            tensor(3.5000)
+            >>> pyhf.tensorlib.percentile(a, 50, axis=1)
+            tensor([7., 2.])
+
+        Args:
+            tensor_in (`tensor`): The tensor containing the data
+            q (`float` or `tensor`): The :math:`q`-th percentile to compute
+            axis (`number` or `tensor`): The dimensions along which to compute
+            interpolation (`str`): The interpolation method to use when the desired
+                percentile lies between two data points ``i < j``:
+
+                - \\'linear\\': ``i + (j - i) * fraction``, where ``fraction`` is the fractional part of the index surrounded by ``i`` and ``j``
+
+                - \\'lower\\': Not yet implemented in PyTorch
+
+                - \\'higher\\': Not yet implemented in PyTorch
+
+                - \\'midpoint\\': Not yet implemented in PyTorch
+
+                - \\'nearest\\': Not yet implemented in PyTorch
+
+        Returns:
+            PyTorch tensor: The value of the :math:`q`-th percentile of the tensor along the specified axis.
+
+        """
+        # Interpolation options not yet supported
+        # c.f. https://github.com/pytorch/pytorch/issues/38349
```

Comment:
should be out with pytorch/pytorch#48523 being done?

Reply:
Nope -- need 1.7.2 (https://github.com/pytorch/pytorch/releases/tag/v1.7.1)

Reply:
Yeah, sadly. I just keep rebasing these PRs so that they don't fall too far behind and make it harder later on.